app/src/main/java/edu/wpi/first/shuffleboard/app/ShuffleboardUpdateChecker.java

```diff
@@ -218,8 +218,11 @@ public static void downloadNewestRelease(URL newestVersionLocation,
   private static File downloadFile(URL remoteFile, DoubleConsumer progressNotifier) throws IOException {
     File temp = File.createTempFile("newestshuffleboard", ".jar");
     log.finer("Downloading to " + temp);
+    URLConnection connection = remoteFile.openConnection();
+    connection.setReadTimeout(5000);    // ms
+    connection.setConnectTimeout(5000); // ms
     try (OutputStream out = new FileOutputStream(temp)) {
-      try (InputStream in = remoteFile.openStream()) {
+      try (InputStream in = connection.getInputStream()) {
```

Comment:
Not related to this PR but is there a reason this was not a try with multiple resources statement?

Reply:
I find it less readable

Reply:
👍